mautrix_instagram/matrix.py

```diff
@@ -122,6 +122,9 @@ async def handle_read_receipt(
         data: SingleReceiptEventContent,
     ) -> None:
         message = await DBMessage.get_by_mxid(event_id, portal.mxid)
+        if not message:
+            # Message might actually be reaction - mark all as read
+            message = await DBMessage.get_last(portal.mxid)
         if not message or message.is_internal:
             return
```

Comment:
The `or message.is_internal` check can be moved to the above `if` (so if the target is an internal item ID, everything is marked as read)